src/test/java/edu/harvard/iq/dataverse/util/SystemConfigTest.java

```diff
@@ -1,13 +1,46 @@
 package edu.harvard.iq.dataverse.util;
 
+import edu.harvard.iq.dataverse.settings.JvmSettings;
+import edu.harvard.iq.dataverse.util.testing.JvmSetting;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 class SystemConfigTest {
 
+    SystemConfig systemConfig = new SystemConfig();
+    
+    @Test
+    void testGetVersion() {
+        // given
+        String version = "100.100";
+        System.setProperty(JvmSettings.VERSION.getScopedKey(), version);
+        
+        // when
+        String result = systemConfig.getVersion(false);
+        
+        // then
+        assertEquals(version, result);
+    }
+    
+    @Test
+    @JvmSetting(key = JvmSettings.VERSION, value = "100.100")
+    @JvmSetting(key = JvmSettings.BUILD, value = "FOOBAR")
+    void testGetVersionWithBuild() {
+        // when
+        String result = systemConfig.getVersion(true);
+    
+        // then
+        assertTrue(result.startsWith("100.100"), "'" + result + "' not starting with 100.100");
+        assertTrue(result.contains("build"));
+        
+        // Cannot test this here - there might be the bundle file present which is not under test control
```

Comment:
A bundle entry would affect the real app the same way, right? So should this test be here to assure the default app is not shipping a bundle file that's going to override this setting? Or should the code prioritize the mpconfig value over any hardcoded bundle entry (that would allow the test to work, but is it also a better choice for the app - should you have to delete a bundle entry to be able to control build number? I'm asking and not recommending - you/others probably have a better sense of whether a bundle entry should have precedence or just be a default.)

Reply:
@poikilotherm my first thought is that this shouldn't be at the top of the community backlog if there's still stuff to figure out! 😄

Reply:
😲 

Why is that? It's the first time this code is getting a review and raised a question from a core member. I was fine with the code as is, it works as designed and tested. Maybe I'm getting the scope of "review" wrong?

Reply:
When I run this test I get this as the result:

    100.100 build pdurbin

This is because I have a file here:

```
$ cat src/main/java/BuildNumber.properties
build.number=pdurbin
```

The test passes just fine, which is good! As a developer I should be free to specify a custom build number.

> So should this test be here to assure the default app is not shipping a bundle file that's going to override this setting? 

No. In fact, we do ship a BuildNumber.properties in each war file. I'm not sure why we would assert that we don't do this but I'm probably misunderstanding what's being discussed above.

In short, I think we should get this into a sprint so core team members can focus on a proper review while it's in a sprint. Again, it sounds like @poikilotherm is ok with the PR as-is. Great.